src/include/sof/trace.h

```diff
@@ -86,46 +86,63 @@
 #define TRACEE	1
 #define TRACEM	0 /* send all trace messages to mbox and local trace buffer */
 
-void _trace_event0(uint32_t log_entry);
-void _trace_event_mbox0(uint32_t log_entry);
-void _trace_event_atomic0(uint32_t log_entry);
-void _trace_event_mbox_atomic0(uint32_t log_entry);
-
-void _trace_event1(uint32_t log_entry, uint32_t param);
-void _trace_event_mbox1(uint32_t log_entry, uint32_t param);
-void _trace_event_atomic1(uint32_t log_entry, uint32_t param);
-void _trace_event_mbox_atomic1(uint32_t log_entry, uint32_t param);
-
-void _trace_event2(uint32_t log_entry, uint32_t param1, uint32_t param2);
-void _trace_event_mbox2(uint32_t log_entry, uint32_t param1, uint32_t param2);
-void _trace_event_atomic2(uint32_t log_entry, uint32_t param1, uint32_t param2);
-void _trace_event_mbox_atomic2(uint32_t log_entry, uint32_t param1,
-	uint32_t param2);
-
-void _trace_event3(uint32_t log_entry, uint32_t param1, uint32_t param2,
-	uint32_t param3);
-void _trace_event_mbox3(uint32_t log_entry, uint32_t param1, uint32_t param2,
-	uint32_t param3);
-void _trace_event_atomic3(uint32_t log_entry, uint32_t param1, uint32_t param2,
-	uint32_t param3);
-void _trace_event_mbox_atomic3(uint32_t log_entry, uint32_t param1,
-	uint32_t param2, uint32_t param3);
-
-void _trace_event4(uint32_t log_entry, uint32_t param1, uint32_t param2,
-	uint32_t param3, uint32_t param4);
-void _trace_event_mbox4(uint32_t log_entry, uint32_t param1, uint32_t param2,
-	uint32_t param3, uint32_t param4);
-void _trace_event_atomic4(uint32_t log_entry, uint32_t param1, uint32_t param2,
-	uint32_t param3, uint32_t param4);
-void _trace_event_mbox_atomic4(uint32_t log_entry, uint32_t param1,
-	uint32_t param2, uint32_t param3, uint32_t param4);
+void _trace_event0(uint32_t log_entry, uint32_t pipe_id, uint32_t comp_id);
```

Comment:
We could make these even more generic and call them id0 and id1. This would mean userspace would know to decode id0 as pipe id and id1 as comp id (given an audio component class). Additionally a macro could be created in pipeline.h and component.h to automatically fill in the pipe and comp ids for users. e.g.
```
trace_comp0(comp, log_entry) \
        _trace_event0(log_entry, comp->pipeline->id, comp->id)
```

Reply:
Such macro is already implemented for pipelines - trace_pipe_with_ids. I will implement similar macros for all trace classes at start of next week.